src/doc-utils.js

```diff
@@ -33,6 +33,23 @@ function traverseDoc(doc, onEnter, onExit) {
   traverseDocRec(doc);
 }
 
+function mapDoc(doc, func) {
+  doc = func(doc);
+
+  if(doc.type === "concat") {
```

Comment:
Subjectively, removing `else`s would make this more readable since you are returning from conditionals.

You use this approach in `removeLines()` so would also have the benefit of keeping style consistent.